### PR TITLE
Bug fix for validation does not prevent the recalculation from occurring bug #8830

### PR DESCRIPTION
**Pull request by ksantiagoBAH**
## Description
Validation in zip code box does not prevent the recalculation from occurring, bug #8830

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/8830

## Testing done
local testing completed

## Acceptance criteria
- [x] Calculation doesn't happen when invalid date is entered
- [x] Learning format and schedule accordion is expanded on error when clicking calculate benefits button

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/gi/containers/EstimateYourBenefits.jsx [hunk 1]
```diff
@@ -90,6 +91,7 @@ export class EstimateYourBenefits extends React.Component {
       <div className="row calculate-your-benefits">
         <EstimateYourBenefitsForm
           profile={profile}
+          calculator={calculator}
```
Comment (zurbergram) [inline src/applications/gi/containers/EstimateYourBenefits.jsx:94]: this duplicates the property `inputs`

File: src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx [hunk 2]
```diff
@@ -1008,6 +1008,9 @@ class EstimateYourBenefitsForm extends React.Component {
   };
 
   render() {
+    const beneficiaryZIPError = this.props.calculator.beneficiaryZIPError;
+    const zipcode = this.props.calculator.beneficiaryZIP;
```
Comment (zurbergram) [inline src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:1012]:
```suggestion
    const beneficiaryZIPError = this.props.inputs.beneficiaryZIPError; 
    const zipcode = this.props.inputs.beneficiaryZIP;
```

File: src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx [hunk 3]
```diff
@@ -1019,7 +1022,14 @@ class EstimateYourBenefitsForm extends React.Component {
         </ul>
         <button
           className="calculate-button"
-          onClick={this.props.updateEstimatedBenefits}
+          onClick={
```
Comment (zurbergram) [inline src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:1025]: can this be pulled out into a variable for easier readability?

Comment (ksantiagomorgan6) [reply src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:1025]: Sure

**Review by zurbergram — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by zurbergram — APPROVED** [review]

File: src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx [hunk 4]
```diff
@@ -155,6 +176,13 @@ class EstimateYourBenefitsForm extends React.Component {
     }
   };
 
+  handleInputFocusByName = name => {
```
Comment (zurbergram) [inline src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:179]: Is this used somewhere?

Comment (ksantiagomorgan6) [reply src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:179]: Doesn't look like it I'll remove it

File: src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx [hunk 5]
```diff
@@ -702,8 +727,9 @@ class EstimateYourBenefitsForm extends React.Component {
           : "Please enter the postal code where you'll take your classes";
 
         amountInput = (
-          <div>
+          <div name="zip-question">
```
Comment (zurbergram) [inline src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:730]:
```suggestion
          <div name="beneficiary-zip-question">
```

**Review by zurbergram — APPROVED** [review]

File: src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx [hunk 6]
```diff
@@ -91,6 +94,27 @@ class EstimateYourBenefitsForm extends React.Component {
...
+    if (
+      this.props.eligibility.giBillChapter === '33' &&
+      (beneficiaryZIPError || zipcode.length !== 5)
+    ) {
+      this.toggleLearningFormatAndSchedule(true);
+      setTimeout(() => {
+        const CheckNameOfElement = checkForEmptyFocusableElement(
+          'beneficiaryZIPCode',
+        );
+        scroller.scrollTo('zip-question', getScrollOptions());
```
Comment (zurbergram) [inline src/applications/gi/components/profile/EstimateYourBenefitsForm.jsx:110]:
```suggestion
        scroller.scrollTo('beneficiary-zip-question', getScrollOptions());
```

**Review by sonntag-adam — APPROVED** [review]

**Pull request merged**